### PR TITLE
docs(effects): rewrite testing

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Part of #1383

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: projects/ngrx.io/content/guide/effects/testing.md [hunk 1]
```diff
@@ -1,300 +1,258 @@
...
 
-Provides a mock test provider of the `Actions` Observable for testing effects. This works well with writing
-marble tests and tests using the `subscribe` method on an Observable. The mock Actions will deliver a new Observable to subscribe to for each test.
+### `provideMockActions`
 
-Details on marble tests and their syntax, as shown in the `hot` and `cold` methods, can be found in [Writing Marble Tests](https://github.com/ReactiveX/rxjs/blob/master/doc/writing-marble-tests.md).
-
-Usage:
+An Effect subscribes to the `Actions` Observable to perform side effects.
+`provideMockActions` provides a mock `Actions` Observable to subscribe to for each test individually.
```
Comment (itayod) [inline projects/ngrx.io/content/guide/effects/testing.md]:
```suggestion
`provideMockActions` provides a mock to the `Actions` Observable and subscribe for each test individually.
```

File: projects/ngrx.io/content/guide/effects/testing.md [hunk 2]
```diff
@@ -1,300 +1,258 @@
...
-      imports: [
-        // any modules needed
-      ],
-      providers: [
-        MyEffects,
-        provideMockActions(() => actions$),
-        // other providers
-      ],
-    });
+Later in the test cases, we assign the `actions$` variable to an Actions stream.
```
Comment (itayod) [inline projects/ngrx.io/content/guide/effects/testing.md]:
```suggestion
Later in the test cases, we assign the `actions$` variable to the `Actions` stream.
```

File: projects/ngrx.io/content/guide/effects/testing.md [hunk 3]
```diff
@@ -1,300 +1,258 @@
...
-      ],
-    });
+// mock Actions and immediately dispatch a GET action
+actions$ = of({ type: 'GET CUSTOMERS' });
 
-    effects = TestBed.get&lt;MyEffects&gt;(MyEffects);
-    metadata = getEffectsMetadata(effects);
-  });
+// mock the service to prevent an HTTP request
+customersServiceSpy.getAllCustomers.and.returnValue(of([...]));
```
Comment (itayod) [inline projects/ngrx.io/content/guide/effects/testing.md]: IMHO the `[...]` might be confusing, maybe you could use `customerList` instead?

File: projects/ngrx.io/content/guide/effects/testing.md [hunk 4]
```diff
@@ -1,300 +1,258 @@
...
 });
 </code-example>
 
-### Effects as functions
+## Examples
+
+### A non-dispatching Effect
 
-Effects can be defined as functions as well as variables. Defining an effect as a function allows you to define default values while having the option to override these variables during tests. This without breaking the functionality in the application.
+Until now we only saw Effects that dispatch an Action and we verified the dispatched action.
```
Comment (itayod) [inline projects/ngrx.io/content/guide/effects/testing.md]:
```suggestion
Until now, we only saw Effects that dispatch an Action and we verified the dispatched action.
```

**Comment by timdeschryver** [comment]
Thanks for the feedback @itayod , I reworded some sentences in the last commit.

**Comment by itayod** [comment]
LGTM! 👍

**Comment by yluijten** [comment]
@timdeschryver Is there also a full example of the source code example?

File: projects/ngrx.io/content/guide/effects/testing.md [hunk 5]
```diff
@@ -1,300 +1,259 @@
...
+  expect(router.navigateByUrl).toHaveBeenCalledWith('customers/bob');
+});
 </code-example>
 
-Within our tests we can now override the default properties:
+### Effect that uses state
+
+Leverage [`MockStore`](/guide/store/testing#using-a-mock-store) and [`MockSelectors`](/guide/store/testing#using-mock-selectors) to test Effects that are selecting slices of the state.
+
+An example of this is to not fetch an entity (customer in this case) when it's already in the store state.
```
[the remark this answers is not in the document]
Comment (timdeschryver) [reply projects/ngrx.io/content/guide/effects/testing.md]: There's an example at https://ngrx.io/guide/effects#incorporating-state.
I didn't add any effects code in the testing guide, only how to test them.
If we add an effects example for this test, we should add them for the rest too.
Thoughts?

File: projects/ngrx.io/content/guide/effects/testing.md [hunk 6]
```diff
@@ -1,300 +1,262 @@
...
+  this.actions$.pipe(
+    ofType(BookActions.search),
+    debounceTime(debounce, scheduler),
+    ...
+  )
+);
+</code-example>
+
+<code-example header="my.effects.spec.ts">
+effects.searchCustomers$({
```
Comment (brandonroberts) [inline projects/ngrx.io/content/guide/effects/testing.md]:
```suggestion
effects.search$({
```

**Comment by timdeschryver** [comment]
@yluijten You can find examples at https://github.com/vsavkin/testing_ngrx_effects and https://github.com/tomastrajan/angular-ngrx-material-starter

**Pull request merged**